f3dex2.s

```diff
@@ -1517,18 +1599,38 @@ ucode equ $11
 status equ $12
 task_yield:
     lw      ucode, OSTask + OSTask_ucode
+.if UCODE_METHOD == METHOD_FIFO
     sw      taskDataPtr, lbl_0BF8
     sw      ucode, lbl_0BFC
-    li      status, SP_SET_SIG1 | SP_SET_SIG2   // yielded and task done signals
+    li      status, SP_SET_SIG1 | SP_SET_SIG2  // yielded and task done signals
     lw      $24, OSTask + OSTask_yield_data_ptr
-    li      $20, -0x8000
+    li      $20, 0x8000
     li      $19, OS_YIELD_DATA_SIZE - 1
+.else // UCODE_METHOD == METHOD_XBUS
+    sw      taskDataPtr, inputBuffer // Is this right? Seems weird.
+    sw      ucode, inputBuffer + 4
```

Comment:
These should be using `OS_YIELD_DATA_SIZE` in some way instead of a data label. I fixed the FIFO version in #4 to use it instead of those two labels. The instructions are different due to the different values for `OS_YIELD_DATA_SIZE` between FIFO and XBUS